kms/src/main/java/kms/CreateKeyRotationSchedule.java

```diff
@@ -0,0 +1,78 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kms;
+
+// [START kms_create_key_rotation_schedule]
+import com.google.cloud.kms.v1.CryptoKey;
+import com.google.cloud.kms.v1.CryptoKey.CryptoKeyPurpose;
+import com.google.cloud.kms.v1.CryptoKeyVersion.CryptoKeyVersionAlgorithm;
+import com.google.cloud.kms.v1.CryptoKeyVersionTemplate;
+import com.google.cloud.kms.v1.KeyManagementServiceClient;
+import com.google.cloud.kms.v1.KeyRingName;
+import com.google.protobuf.Duration;
+import com.google.protobuf.Timestamp;
+import java.io.IOException;
+import java.time.temporal.ChronoUnit;
+
+public class CreateKeyRotationSchedule {
+
+  public void createKeyRotationSchedule() throws IOException {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "your-project-id";
+    String locationId = "us-east1";
+    String keyRingId = "my-key-ring";
+    String id = "my-key";
+    createKeyRotationSchedule(projectId, locationId, keyRingId, id);
+  }
+
+  // Create a new key that automatically rotates on a schedule.
+  public void createKeyRotationSchedule(
+      String projectId, String locationId, String keyRingId, String id) throws IOException {
+    // Initialize client that will be used to send requests. This client only
+    // needs to be created once, and can be reused for multiple requests. After
+    // completing all of your requests, call the "close" method on the client to
+    // safely clean up any remaining background resources.
+    try (KeyManagementServiceClient client = KeyManagementServiceClient.create()) {
+      // Build the parent name from the project, location, and key ring.
+      KeyRingName keyRingName = KeyRingName.of(projectId, locationId, keyRingId);
+
+      // Calculate the date 24 hours from now (this is used below).
+      long tomorrow = java.time.Instant.now().plus(24, ChronoUnit.HOURS).getEpochSecond();
+
+      // Build the key to create with a rotation schedule.
+      CryptoKey key =
+          CryptoKey.newBuilder()
+              .setPurpose(CryptoKeyPurpose.ENCRYPT_DECRYPT)
+              .setVersionTemplate(
+                  CryptoKeyVersionTemplate.newBuilder()
+                      .setAlgorithm(CryptoKeyVersionAlgorithm.GOOGLE_SYMMETRIC_ENCRYPTION))
+
+              // Rotate every 30 days.
+              .setRotationPeriod(
+                  Duration.newBuilder().setSeconds(java.time.Duration.ofHours(24).getSeconds()))
```

Comment:
This says "Rotate every 30 days", but gives a duration of 24 hours. Is that correct?

Reply:
Good catch - fixed